lib/render/FlowRenderer.cpp

```diff
@@ -322,17 +345,257 @@ FlowRenderer::_paintGL( bool fast )
     glDepthMask(true);
 
     _prepareColormap( params );
-    _renderFromAnAdvection( &_advection, params, fast );
-    /* If the advection is bi-directional */
-    if( _2ndAdvection )
-        _renderFromAnAdvection( _2ndAdvection.get(), params, fast );
+    
+    int ret = 0;
+    
+    if (params->GetValueLong("old_render", 0)) {
+        _renderFromAnAdvectionLegacy( &_advection, params, fast );
+        /* If the advection is bi-directional */
+        if( _2ndAdvection )
+            _renderFromAnAdvectionLegacy( _2ndAdvection.get(), params, fast );
+    } else {
+        // Workaround for how bi-directional was implemented.
+        if (_2ndAdvection)
+            _renderStatus = FlowStatus::SIMPLE_OUTOFDATE;
```

Comment:
Does this line duplicate with Line 364?

Reply:
The value is reset by `_renderAdvection`

Reply:
Let me try again.

if you remove the current line 358 and 359, will the execution be different?

Reply:
Yes.

Reply:
Ugh, if it were me I'd put a better explanation than a single-worded "Yes."

Overall, it's more of your interest than anyone else's interest to have this PR resolved and merged, isn't it?

Reply:
A simple comment that _renderAdvection resets _2ndAdvection would clarify what is happening here.

Reply:
Done.

Reply:
@clyne It doesn't seem your description is accurate. The function `_renderAdvection()` takes in an advection pointed by a const pointer, and renders whatever that Advection provides. How can a function reset a const pointer?

Reply:
@shaomeng see comments added by Stas'

Reply:
@clyne I saw that but I don't think my question is answered in either comments or this conversation list. Let me try again: 

In the current code, line 362 and 367 perform the same test, but I don't see anything could possibly change between these two lines that results in different test results. This makes me think if removing the test at line 362 would actually change the execution from the current form. If the answer is yes, I'd appreciated a more detailed explanation on where and under what condition the execution would change.

Reply:
It's not the test results that will change, it's the behavior of _renderAdvection that will change because it references  _renderStatus

Reply:
Thanks for the explanation John. It makes more sense now, but the logic still seems confusing: the rendering of the first advection (line 365) is dependent on if the second advection exists or not (line 362-363) ?? 

Stas, I'm curious why it is designed this way?